File: Pattern/Two Pointers/167: Two Sum II - Input Array Is Sorted/Solution.js
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (numbers, target) {
  let len = numbers.length;
  let left = 0;
  let right = len - 1;
  while (left < right) {
    let sum = numbers[left] + numbers[right];
    if (sum === target) {
      return [++left, ++right];
    } else if (sum < target) {
      left++;
    } else {
      right--;
    }
  }
  return [];
};

File: pattern/Two Pointers/167: Two Sum II - Input Array Is Sorted/Solution.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        begin = 0
        end = len(numbers) - 1

        while begin < end:
            currentSum = numbers[begin] + numbers[end]
            if currentSum == target:
                return [begin + 1, end + 1]   # 1-indexed
            elif currentSum > target:
                end -= 1
            else:
                begin += 1
        return []